src/components/compress/index.tsx

```diff
@@ -240,6 +240,7 @@ export default class Compress extends Component<Props, State> {
   private readonly encodeCache = new ResultCache();
   private readonly leftProcessor = new Processor();
   private readonly rightProcessor = new Processor();
+  private readonly updateImageTimeoutIds: [number, number] = [0, 0];
```

Comment:
My hunch is that these 2 zeroes represent the imaginary timeout IDs of both processors, to not have optional numbers in TypeScript here? I think it would be clearer to make this an object instead or else add a clarifying comment.